### ISSUE TITLE
Price bucket calculation for Custom Price Granularity

### ISSUE THREAD

**Issue by Pubmatic-Dhruv-Sonone** [prebid/prebid-server#2469]
While using custom price granularity, prebid server return pb values lesser than min value of bucket
Example : For  the CPM =2 and  price granularity:
```
 {
 	Min:       0.0,
 	Max:       1.5,
 	Increment: 1.0,
 },
 {
 	Min:       1.5,
 	Max:       2.5,
 	Increment: 1.2,
 }
```
I expected pb value to be 1.5 but prebid-server calculates it as 1.2.
why is min bucket value not being taken into consideration for computing pb value?

Also Prebid-server seems to be using  different logic than prebid.js for price bucket calcuation.

Below is the logic used by Prebid.js.
```
var pow = Math.pow(10, precision + 2);
var cpmToFloor = (cpm * pow - bucketMin * pow) / (increment * pow);
var cpmTarget = Math.floor(cpmToFloor) * increment + bucketMin;
```

**Comment by bretg** [comment]
You misunderstand min and max.

Custom price granularities are defined as ranges. So another way to write the example you posted is

- for prices between 0 and 1.5, round down to the nearest multiple of 1.0
- for prices between 1.5 and 2.5, round down to the nearest multiple of 1.2

Prebid Server is doing it correctly in this case, and I believe it's in sync with what Prebid.js is doing except that the javascript now supports an arbitrary javascript function for rounding, which is of course impossible in server. If you have an example where there's a difference, please post the details.

Also worth noting that "min" is not supported. See https://docs.prebid.org/prebid-server/endpoints/openrtb2/pbs-endpoint-auction.html#targeting

```
Top end of this range of price buckets. The bottom end is 0 or the max of the previous bucket. Note: in order to prevent ranges with gaps, there’s no ‘min’ attribute.
```

**Comment by bretg** [comment]
Ok, you're right that there's a difference. The Prebid.js cpmBucketManager.js implements getCpmStringValue saying: "start increments at the bucket min"

Do you have a real-world example where this causes a problem? We'd like to use it as another test case.

Since this difference has existed for 5 years, it's hard for me to consider this an urgent issue. We'll fix it, but can't currently promise a timeline.

To the PBS-Go and Java developers who take this on, here's the goal: make PBS resolve hb_pb the same way as Prebid.js.

With this config:
```
{
  "ext": {
    "prebid": {
      "targeting": {
        "pricegranularity": {
          "precision": 2,
          "ranges": [{
            "max": 1.50,
            "increment": 1.0
          },{
            "max": 2.50,
            "increment": 1.2
          }]
        },
        ...
    }
}
```
A bid price of 2 should resolve the hb_pb value to be 1.5 but prebid-server currently calculates it as 1.2.

**Comment by SyntaxNode** [comment]
Fixed in PBS-Go 0.240.0.

**Comment by bretg** [comment]
Fixed in PBS-Java 2.6
